sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4118,6 +4118,26 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     }
   }
 
+  test("try_cast") {
+    Seq("true", "false").foreach { ansiEnabled =>
+      withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiEnabled) {
```

Comment:
It looks this expr itself does not depend on the ansi option, but we need tests for ansi=true?

Reply:
Yes it doesn't depend on ANSI option. We need to make sure that it doesn't throw exception when ansi=true